oauthenticator/github.py

```diff
@@ -215,8 +231,68 @@ async def authenticate(self, handler, data=None):
                     auth_state['github_user']['email'] = val['email']
                     break
 
+        if self.fetch_teams:
+            if 'read:org' not in self.scope:
+                # This means the 'read:org' scope was not set, and we can't fetch teams
+                self.log.error(
+                    'read:org scope is required for fetch_teams functionality to work'
+                )
+            else:
+                # Number of teams to request per page
+                per_page = 100
+
+                # Fetch all teams across all orgs this user is in and we have visibility for
+                # https://docs.github.com/en/rest/reference/teams#list-teams-for-the-authenticated-user
```

Comment:
This kind of clarity is something I think belongs to the docstring of the configuration to enable this. I had to ask myself what it meant to fetch teams as part of a user rather than an organization etc.

Also, since it notes "we have visibility for" I become hesitant on my understanding - wouldn't that be all teams if we have requested `read:org`? Or is the complexity like partial visibility because both the user part of the team and the github org needs to agree on this being made available?

Reply:
I tried to find docs on GitHub's API what the 'Grant' button that shows up sometimes when i log in for some organizations, and I don't actually know why exactly. Given this is the exact situation current feature wrt allowing only users with specific teams to log in, I'm just going to remove the ambiguous comment and let that be.